wps/services.py

```diff
@@ -0,0 +1,164 @@
+        client_id = settings.SPOTIFY_CLIENT_ID
+        client_secret = settings.SPOTIFY_CLIENT_SECRET
+        b64_client = base64.b64encode(
+            f'{client_id}:{client_secret}'.encode()
+        ).decode()
+        headers = {'Authorization': f'Basic {b64_client}'}
+        data = {'grant_type': 'client_credentials'}
+        url = 'https://accounts.spotify.com/api/token'
+        response = requests.post(url, data=data, headers=headers, timeout=30)
+        if response.status_code != 200:
+            raise requests.exceptions.RequestException(response.text)
+        access_token = response.json().get('access_token')
+        if not access_token:
+            raise requests.exceptions.RequestException(
+                f'Failed to retrieve access token, '
+                f'code: {response.status_code}, response: {response.text}'
+            )
+        return access_token
+
+    @staticmethod
+    def _format_recommendation_response(
+        response: dict,
+    ) -> List[Recommendation]:
+        if not response or not response.get('tracks'):
+            raise NoContentFound(
+                f'Invalid or missing content in response {response}'
+            )
+        formatted_data = []
+        recommendations = response['tracks']
+        for recommendation in recommendations:
+            formatted_data.append(
+                {
+                    'playlist': recommendation['album']['name'],
+                    'url': recommendation['external_urls']['spotify'],
+                    'artists': [
+                        artist['name'] for artist in recommendation['artists']
+                    ],
+                    'total_tracks': round(
+                        recommendation['album']['total_tracks']
+                    ),
+                    'duration_seconds': round(
+                        recommendation['duration_ms'] / 60, 2
+                    ),
+                }
+            )
+        return formatted_data
+
+    def get_recommendations(self, genres: Union[List[str], str]):
+        url = self.base_url + '/recommendations'
+        if isinstance(genres, list):
+            genres = str(genres).replace(' ', '')[1:-1]
+        params = {
+            'limit': self.recommendation_limit,
+            'seed_genres': genres,
+        }
+        response = requests.get(
+            url, params=params, headers=self.headers, timeout=30
+        )
+        if response.status_code != 200:
+            raise requests.exceptions.RequestException(response.text)
+        return self._format_recommendation_response(response.json())
+
+
+def get_playlist_api():
+    return SpotifyAPI()
+
+
+def get_genre(temperature: float) -> str:
+    if temperature > 30:
+        return 'party'
+    if temperature > 14:
+        return 'pop'
+    if temperature > 10:
+        return 'rock'
+    return 'classical'
+
+
+def get_playlists_recommendations(location: dict) -> List[Recommendation]:
+    openweather_api = get_weather_api()
```

Comment:
If you are using a factory method here, your variable should also be named with a generic name